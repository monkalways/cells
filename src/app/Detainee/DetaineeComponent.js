import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core';
import queryString from 'query-string';
import Layout from '../common/Layout';
import Header from './Header';
import Footer from './Footer';
import DetaineeDetails from './DetaineeDetails';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  getDetainee: PropTypes.func.isRequired,
  getAvailableActivityRooms: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }).isRequired,
  logOut: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

class DetaineeComponent extends Component {
  componentDidMount() {
    const { match } = this.props;
    const { id } = match.params;
    const { getDetainee, getAvailableActivityRooms } = this.props;
    getDetainee(id);
    getAvailableActivityRooms();
  }

  handleLogout = () => {
    const { logOut, location } = this.props;
    const { first, second } = queryString.parse(location.search);
    logOut(first, second);
  };

  render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <Layout>
          <Header onLogout={this.handleLogout} />
          <div className={classes.body}>
            <DetaineeDetails />
          </div>
          <Footer />
        </Layout>
      </React.Fragment>
    );
  }
}

DetaineeComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  body: {
    marginTop: theme.spacing.unit,
  },
}))(DetaineeComponent);
