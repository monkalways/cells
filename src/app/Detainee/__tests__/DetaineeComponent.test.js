import React from 'react';
import { shallow } from 'enzyme';

import { DetaineeComponent } from '../DetaineeComponent';

describe('DetaineeComponent', () => {
  let initialize;
  let push; // history
  let search; // location
  let id; // match
  let logOut;

  const first = 'cells';
  const second = 'B4';

  beforeEach(() => {
    initialize = jest.fn();
    push = jest.fn();
    search = `first=${first}&second=${second}`;
    id = '1';
    logOut = jest.fn();
  });

  const setup = () => {
    const classes = {
      body: 'body',
    };

    return shallow(<DetaineeComponent
      classes={classes}
      initialize={initialize}
      history={{ push }}
      location={{ search }}
      logOut={logOut}
      match={{
        params: {
          id,
        },
      }}
    />);
  };

  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call initialize when componentDidMount is called', () => {
    const wrapper = setup();

    wrapper.instance().componentDidMount();

    expect(initialize).toBeCalledWith(id);
  });

  it('should call logOut when handleLogout is called', () => {
    const wrapper = setup();

    wrapper.instance().handleLogout();

    expect(logOut).toBeCalledWith(first, second);
  });
});
