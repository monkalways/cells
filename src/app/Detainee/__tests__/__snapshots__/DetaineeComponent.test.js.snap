// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetaineeComponent should render component 1`] = `
<Fragment>
  <WithStyles(LayoutComponent)>
    <Connect(WithStyles(HeaderComponent))
      onLogout={[Function]}
    />
    <div
      className="body"
    >
      <Connect(DetaineeDetailsComponent) />
    </div>
    <WithStyles(withRouter(FooterComponent)) />
  </WithStyles(LayoutComponent)>
</Fragment>
`;
